gen/insight_mu

```diff
@@ -0,0 +1,141 @@
+my $ampm = "AM";
+if ($hh > 12) {
+	$ampm = "PM";
+}
+my $writeTime = strftime ("%d/%m/%Y %I:%M ", gmtime) . $ampm;
+
+my $memberDataById;
+my $groupDataById;
+my $workplaceDataById;
+foreach my $resourceId ($data->getResourceIds()) {
+	foreach my $groupId ($data->getGroupIdsForResource(resource => $resourceId)) {
+		my $groupName = $data->getGroupAttributeValue(group => $groupId, attrName => $A_GROUP_NAME);
+		my $groupDesc = $data->getGroupAttributeValue(group => $groupId, attrName => $A_GROUP_DESC);
+		my $groupCISPR = $data->getGroupAttributeValue(group => $groupId, attrName => $A_GROUP_INETCISPR);
+		my $groupVAZPR = $data->getGroupAttributeValue(group => $groupId, attrName => $A_GROUP_INETVAZPR);
+		my @o365emails = $data->getGroupAttributeValue(group => $groupId, attrName => $A_GROUP_O365_EMAIL_ADDRESS);
+		my $o365groupName = $data->getGroupAttributeValue(group => $groupId, attrName => $A_GROUP_O365_NAME);
+		my @members = ();
+		for my $memberId ($data->getMemberIdsForResourceAndGroup(resource => $resourceId, group => $groupId)) {
+			my $memberStatus = $data->getMemberAttributeValue( member => $memberId, attrName => $A_MEMBER_STATUS  );
+			# process only valid members
+			next if (! $memberStatus eq $STATUS_VALID);
+			my $userId = $data->getUserIdForMember (member => $memberId);
+			push(@members, $data->getUserAttributeValue(member => $memberId, attrName => $A_USER_UCO));
+			my $uco = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_UCO );
+			my $firstName = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_FIRSTNAME );
+			my $lastName = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_LASTNAME );
+			my $person = {
+				Name                => $firstName . " " . $lastName . " (" . $uco . ")",
+				UCO                 => $uco,
+				ExternalObjectID    => $userId,
+				FirstName           => $firstName,
+				LastName            => $lastName,
+				EmailAddress        => $uco . "\@muni.cz",
+				Expired 			=> "FALSE",
+				LastImportWriteTime => $writeTime
+			};
+			$memberDataById->{$memberId} = $person;
+		}
+		# decide whether group or workplace depending on defined VAZPR attribute
+		if ( ! defined $groupVAZPR || $groupVAZPR eq '' ) {
+			my $group = {
+				Name                => $groupName,
+				Description         => $groupDesc,
+				ExternalObjectId    => $groupId,
+				EmailAddress        => "",
+				Expired 			=> "FALSE",
+				LastImportWriteTime => $writeTime,
+				Members             => \@members
+			};
+			$groupDataById->{$groupId} = $group;
+		} else {
+			if ( defined $o365groupName && ! $o365groupName eq '' ) {
+				$groupName = $o365groupName;
+			}
+			my $group = {
+				Name                => $groupName . " (" . $groupCISPR . "/" . $groupId . ")",
+				Description         => $groupDesc,
+				InetVAZPR           => $groupVAZPR,
+				InetCISPR           => $groupCISPR,
+				ExternalObjectID    => $groupId,
+				EmailAddress        => $o365emails[0][0],
```

Comment:
The reason is simple. You are getting a reference to the array, not the array itself. What you can do is something like this:
`my @array = $getArrayFromPerun() ? @($getArrayFromPerun()) : ();`
It will check if there is something in the reference and if yes, you will retype it to the array. If not, you will return an empty array on your own. Then You can use "$o365emails[0]" properly.